docs/workflow/trimming/feature-switches.md

```diff
@@ -8,18 +8,20 @@ configurations but their defaults might vary as any SDK can set the defaults dif
 
 | MSBuild Property Name | AppContext Setting | Description |
 |-|-|-|
+| CustomResourceTypesSupport | System.Resources.ResourceManager.AllowCustomResourceTypes | Use of custom resource types is disabled when set to false. ResourceManager code paths that use reflection for custom types can be trimmed. |
 | DebuggerSupport | System.Diagnostics.Debugger.IsSupported | Any dependency that enables better debugging experience to be trimmed when set to false |
+| EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization | System.ComponentModel.TypeConverter.EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization | BinaryFormatter serialization support is trimmed when set to false. |
 | EnableUnsafeUTF7Encoding | System.Text.Encoding.EnableUnsafeUTF7Encoding | Insecure UTF-7 encoding is trimmed when set to false |
 | EnableUnsafeBinaryFormatterSerialization | System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization | BinaryFormatter serialization support is trimmed when set to false |
 | EventSourceSupport | System.Diagnostics.Tracing.EventSource.IsSupported | Any EventSource related code or logic is trimmed when set to false |
 | InvariantGlobalization | System.Globalization.Invariant | All globalization specific code and data is trimmed when set to true |
-| UseSystemResourceKeys | System.Resources.UseSystemResourceKeys |  Any localizable resources for system assemblies is trimmed when set to true |
 | HttpActivityPropagationSupport | System.Net.Http.EnableActivityPropagation | Any dependency related to diagnostics support for System.Net.Http is trimmed when set to false |
 | UseNativeHttpHandler | System.Net.Http.UseNativeHttpHandler | HttpClient uses by default platform native implementation of HttpMessageHandler if set to true. |
 | StartupHookSupport | System.StartupHookProvider.IsSupported | Startup hooks are disabled when set to false. Startup hook related functionality can be trimmed. |
+| UseSystemResourceKeys | System.Resources.UseSystemResourceKeys |  Any localizable resources for system assemblies is trimmed when set to true |
 | TBD | System.Threading.ThreadPool.EnableDispatchAutoreleasePool | When set to true, creates an NSAutoreleasePool around each thread pool work item on applicable platforms. |
-| CustomResourceTypesSupport | System.Resources.ResourceManager.AllowCustomResourceTypes | Use of custom resource types is disabled when set to false. ResourceManager code paths that use reflection for custom types can be trimmed. |
-| EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization | System.ComponentModel.TypeConverter.EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization | BinaryFormatter serialization support is trimmed when set to false. |
+| _EnableCallingManagedFunctionFromNative | Internal.Runtime.InteropServices.ComponentActivator.IsSupported | Getting a managed function from native hosting is disabled when set to false and related functionality can be trimmed. |
```

Comment:
Why do these thave underscores in their MSBuild names? If these are not considered public switches, we shouldn't be documenting them.

Reply:
My thinking was that by adding them to the doc in the runtime repo, we're not actually documenting them for public use (just like any other internal feature description). But I would not mind too much making these non-underscore. I just don't think we should actually document these on docs - at least not yet.

Reply:
Since the MSBuild property is just a convenience on top of setting the `RuntimeHostConfigurationOption`, I would suggest we don't create any MSBuild property if we don't intend for people to use it.

Reply:
Side question: what's the benefit if having an internal feature switch that we don't want people to use?

Reply:
Unfortunately I already know about couple of users who will want to enable this feature switch - it's absolutely unsupported, but that's a different story. I didn't want to add a hard breaking change for these - and thus we need some SDK mechanism to enable/disable this -> feature switch.

The reason I don't feel comfortable to fully document this is that trimming native hosting targets is currently weird. Typically the native hosting target is a classlib - so not self-contained -> no trimming. The only use case for this is a very special scenario in the native hosting APIs which lets you use self-contained app like that, which can be trimmer potentially.

Reply:
>  and thus we need some SDK mechanism to enable/disable this -> feature switch

They can do that by setting the correct `RuntimeHostConfigurationOption` in the project. e.g.

```xml
  <ItemGroup>
    <RuntimeHostConfigurationOption Include="Internal.Runtime.InteropServices.ComponentActivator.IsSupported"
                                    Value="true"
                                    Trim="true" />
```

There's no need to define a "private" MSBuild property, or even add this to the sdk, if we don't want this to be public.

Reply:
Well SDK has to set this to "false" when trimming is turned on. The question is if it's possible to overwrite it via `csproj` - I don't know.

Reply:
Ok, if we are going to use the property ourselves then I'm fine with creating an MSBuild property for this.